sbtplugin/src/main/scala/scala/scalanative/sbtplugin/ScalaNativePluginInternal.scala

```diff
@@ -108,8 +112,12 @@ object ScalaNativePluginInternal {
       IO.createDirectory(target)
       unpackRtlib(classpath)
       val links = compileNir(opts).map(_.name)
-      compileLl(clang, target, appll, binary, links, linkAs, clangOpts)
-      Process(abs(binary)).!
+      compileLl(clang, target, appll, binary, links, linkage, clangOpts)
+    },
+
+    run := {
+      nativeLink.value
+      Process(abs(nativeBinary.value)).!
```

Comment:
It will do that by default if `nativeLink` is a `taskKey`, which it should be anyway.